src/telliot_ampl_feeds/scripts/submit_uspce.py

```diff
@@ -0,0 +1,147 @@
+"""Submits three month rolling average of the USPCE to TellorX on Rinkeby."""
+import asyncio
+from typing import Optional
+
+from telliot.apps.telliot_config import TelliotConfig
+from telliot.contract.contract import Contract
+from telliot.contract.gas import fetch_gas_price
+from telliot.queries.legacy_query import LegacyRequest
+from telliot.utils.abi import rinkeby_tellor_master
+from telliot.utils.abi import rinkeby_tellor_oracle
+from telliot.utils.response import ResponseStatus
+
+
+def get_cfg() -> TelliotConfig:
+    """Get rinkeby endpoint from config
+
+    If environment variables are defined, they will override the values in config files
+    """
+    cfg = TelliotConfig()
+
+    # Override configuration for rinkeby testnet
+    cfg.main.chain_id = 4
+
+    _ = cfg.get_endpoint()
+
+    return cfg
+
+
+def get_master(cfg: TelliotConfig) -> Optional[Contract]:
+    """Helper function for connecting to a contract at an address"""
+    endpoint = cfg.get_endpoint()
+    if not endpoint:
+        print("Could not connect to master contract.")
+        return None
+
+    endpoint.connect()
+    master = Contract(
+        address="0x657b95c228A5de81cdc3F85be7954072c08A6042",
+        abi=rinkeby_tellor_master,  # type: ignore
+        node=endpoint,
+        private_key=cfg.main.private_key,
+    )
+    master.connect()
+    return master
+
+
+def get_oracle(cfg: TelliotConfig) -> Optional[Contract]:
+    """Helper function for connecting to a contract at an address"""
+    endpoint = cfg.get_endpoint()
+    if not endpoint:
+        print("Could not connect to master contract.")
+        return None
+
+    if endpoint:
+        endpoint.connect()
+    oracle = Contract(
+        address="0x07b521108788C6fD79F471D603A2594576D47477",
+        abi=rinkeby_tellor_oracle,  # type: ignore
+        node=endpoint,
+        private_key=cfg.main.private_key,
+    )
+    oracle.connect()
+    return oracle
+
+
+def parse_user_val() -> int:
+    """Parse USPCE value from user input."""
+    print("Enter USPCE value (example: 13659.3:")
+
+    uspce = None
+
+    while uspce is None:
+        inpt = input()
+
+        try:
+            inpt = int(float(inpt) * 1000000)  # type: ignore
+        except ValueError:
+            print("Invalid input. Enter int or float.")
+            continue
+
+        print(f"Submitting value: {inpt}\nPress [ENTER] to confirm.")
+        _ = input()
+
+        uspce = inpt
+
+    return uspce
+
+
+async def submit(
```

Comment:
this is now encapsulated in the interval reporter

Reply:
I shall update when we merge your PR.